### PR TITLE
Reference Radiation Dev Testing Bugs

### PR DESCRIPTION
**Pull request by demariadaniel**
**Description of changes**

- Fixes incorrect validation logic + tests, were firing too often

**Type of Change**

- [x] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

**Checklist before requesting review:**

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/submission/validation-clinical/therapy.ts [hunk 1]
```diff
@@ -169,13 +169,22 @@ const validateRadiationRecords = async (
     }
 
     // Therapy + Treatment are associated with Radiation
-    const currentTreatmentIsRadiation = treatment_type === 'radiation';
+    const currentTreatmentIsRadiation =
+      treatment_type &&
+      (treatment_type === 'radiation' ||
+        (Array.isArray(treatment_type) && treatment_type.includes('Radiation therapy')));
```
Comment (demariadaniel) [inline src/submission/validation-clinical/therapy.ts:175]: This I need to give a review of treatment_type `string` vs `array` cases

Comment (demariadaniel) [reply src/submission/validation-clinical/therapy.ts]: Removed string cases

File: src/submission/validation-clinical/therapy.ts [hunk 2]
```diff
@@ -169,13 +169,20 @@ const validateRadiationRecords = async (
...
+    const currentTreatmentIsRadiation =
+      treatment_type &&
+      Array.isArray(treatment_type) &&
+      treatment_type.includes('Radiation therapy');
 
     const submittedTreatmentIsRadiation =
-      submissionTreatmentIdMatch && submissionTreatmentIdMatch.treatment_type === 'radiation';
+      submissionTreatmentIdMatch &&
+      Array.isArray(submissionTreatmentIdMatch.treatment_type) &&
+      submissionTreatmentIdMatch.treatment_type.includes('Radiation therapy');
```
Comment (ciaranschutte) [inline src/submission/validation-clinical/therapy.ts:180]: I think both of these can be shortened to remove the first undefined check, as `Array.isArray(undefined) // = false`

File: src/submission/validation-clinical/therapy.ts [hunk 3]
```diff
@@ -199,30 +206,31 @@ const validateRadiationRecords = async (
...
 
     const storedTreatmentDonorIdMatch =
-      storedTreatmentIdMatch?.clinicalInfo?.submittedTherapyDonorId === relatedTreatmentDonorId;
+      storedTreatmentIdMatch?.clinicalInfo?.submittedTherapyDonorId === submittedTherapyDonorId;
 
-    const previousTreatmentDonorId =
-      relatedTreatmentDonorId ||
-      submissionTreatmentIdMatch?.submitter_donor_id ||
-      storedTreatmentIdMatch?.clinicalInfo?.submittedTherapyDonorId;
+    const previousTreatmentDonorIdMatch =
```
Comment (ciaranschutte) [inline src/submission/validation-clinical/therapy.ts:214]: @demariadaniel what does `previousTreatmentDonorId` refer to?

**Review by ciaranschutte — CHANGES_REQUESTED** [review]
small suggestion + plus a q

File: src/submission/validation-clinical/therapy.ts (hunk 3 above)
Comment (demariadaniel) [reply src/submission/validation-clinical/therapy.ts:214]: We compare the currently submitted therapy record's treatment ID against 
a) other treatment on the current donor `treatmentDonorIdMatch` 
b) other treatment records that were submitted `submittedTreatmentDonorIdMatch` 
c) other treatment records in the db `storedTreatmentDonorIdMatch`

File: src/submission/validation-clinical/therapy.ts (hunk 2 above)
Comment (demariadaniel) [reply src/submission/validation-clinical/therapy.ts]: Yes and replaced w/ `?` operators excellent

Comment (demariadaniel) [reply src/submission/validation-clinical/therapy.ts]: Updated

File: src/submission/validation-clinical/therapy.ts [hunk 4]
```diff
@@ -169,13 +169,16 @@ const validateRadiationRecords = async (
...
 
     const submittedTreatmentIsRadiation =
-      submissionTreatmentIdMatch && submissionTreatmentIdMatch.treatment_type === 'radiation';
+      Array.isArray(submissionTreatmentIdMatch?.treatment_type) &&
+      submissionTreatmentIdMatch?.treatment_type.includes('Radiation therapy');
 
     const storedTreatmentIsRadiation =
-      storedTreatmentIdMatch?.clinicalInfo?.treatment_type === 'radiation';
+      Array.isArray(storedTreatmentIdMatch?.clinicalInfo?.treatment_type) &&
+      storedTreatmentIdMatch?.clinicalInfo?.treatment_type.includes('Radiation therapy');
```
Comment (ciaranschutte) [inline src/submission/validation-clinical/therapy.ts:181]: suggestion that we change this for a single object lookup
something like: 
```javascript
const storedTreatmentType = storedTreatmentIdMatch?.clinicalInfo?.treatment_type;
const storedTreatmentIsRadiation =
   Array.isArray(storedTreatmentType) &&
   storedTreatmentType.includes('Radiation therapy');
   ```

**Comment by ciaranschutte** [comment]
@demariadaniel one more cleanup suggestion on the last commit

File: src/submission/validation-clinical/therapy.ts (hunk 4 above)
Comment (demariadaniel) [reply src/submission/validation-clinical/therapy.ts:181]: Sure, will do the same for `submittedTreatmentType`

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**